src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/Program.cs

```diff
@@ -69,16 +79,136 @@ private void DoTheRun(int iterations)
+            RecyclablePooledFactoriesAndChannelsAsync_OpenOnce<WcfService1.IService1, HelloWorldTest>.RunAllScenariosWithWeightsAsync(1000, 1, 1).Wait();
+            //RecyclablePooledFactoriesAndChannelsAsync_OpenOnce<WcfService1.IDuplexService, DuplexTest>.RunAllScenariosWithWeightsAsync(1000, 1, 1).Wait();
+            //RecyclablePooledFactoriesAndChannelsAsync_OpenOnce<WcfService1.IStreamingService, StreamingTest>.RunAllScenariosWithWeightsAsync(1000, 1, 1).Wait();
+
+            // Add more async tests to achieve parity with sync tests
+        }
+
+        // We need a better grouping of tests per binding/feature/stress scenario.
+        // But for now we simply group them in 2 methods below for Http&Net.Tcp binding.
+        // The switch between the protocol being used is in the call to TestHelpers.SetHostAndProtocol(useHttp: true/false, ...)
+
+        // This is a copy of RunAllStressTests with the tests that are not supported by Http binding commented out
+        public static void RunAllHttpStressTests()
+        {
+            //A full cycle of create/use/destroy
+            CreateAndCloseFactoryAndChannelFullCycleTest<WcfService1.IService1, HelloWorldTest>.CreateFactoriesAndChannelsUseAllOnceCloseAll();
+            //CreateAndCloseFactoryAndChannelFullCycleTest<WcfService1.IDuplexService, DuplexTest>.CreateFactoriesAndChannelsUseAllOnceCloseAll();
+            CreateAndCloseFactoryAndChannelFullCycleTest<WcfService1.IStreamingService, StreamingTest>.CreateFactoriesAndChannelsUseAllOnceCloseAll();
+
+            // Pool channel factories and always create/destroy channels
+            PooledFactories<WcfService1.IService1, HelloWorldTest>.CreateUseAndCloseChannels();
+            //PooledFactories<WcfService1.IDuplexService, DuplexTest>.CreateUseAndCloseChannels();
+            PooledFactories<WcfService1.IStreamingService, StreamingTest>.CreateUseAndCloseChannels();
+
+            // Pool channel factories, always create/destroy channels, recycle factories while using them
+            RecyclablePooledFactories<WcfService1.IService1, HelloWorldTest>.RunAllScenariosWithWeights(100, 1);
+            //RecyclablePooledFactories<WcfService1.IDuplexService, DuplexTest>.RunAllScenariosWithWeights(100, 1);
+            RecyclablePooledFactories<WcfService1.IStreamingService, StreamingTest>.RunAllScenariosWithWeights(100, 1);
+
+            // Pool both channel factories and channels, never recycle
+            PooledFactoriesAndChannels<WcfService1.IService1, HelloWorldTest>.UseChannelsInPooledFactoriesAndChannels();
+            //PooledFactoriesAndChannels<WcfService1.IDuplexService, DuplexTest>.UseChannelsInPooledFactoriesAndChannels();
+            PooledFactoriesAndChannels<WcfService1.IStreamingService, StreamingTest>.UseChannelsInPooledFactoriesAndChannels();
+
+
+            //// These will get stuck with wcf/issues/108
+            //// So we comment them out and use *_OpenOnce workaround instead
```

Comment:
Thoughts for the future -- if possible consider making each of these a separate test and allow xunit to be the mechanism that runs them.  We already have an [ActiveIssue(###)] attribute to disable tests known to fail.